### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Triangulated): homological functors

### PR DESCRIPTION
**Pull request by joelriou**
A functor `F : C ⥤ A` from a pretriangulated category to an abelian category is homological iff it sends distinguished triangles in `C` to exact sequences in `A`. In this PR, we define the corresponding type class `F.IsHomological`. If `F` is a homological functor, we introduce the strictly full triangulated subcategory `F.homologicalKernel`.

---
- [x] depends on: #11740

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Triangulated/Functor.lean [hunk 1]
```diff
@@ -116,6 +116,25 @@ lemma map_distinguished [F.IsTriangulated] (T : Triangle C) (hT : T ∈ distTria
     F.mapTriangle.obj T ∈ distTriang D :=
   IsTriangulated.map_distinguished _ hT
 
+namespace IsTriangulated
+
+open ZeroObject
+
+variable [F.IsTriangulated]
+
+instance : PreservesZeroMorphisms F where
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Triangulated/Functor.lean:125]: Should this priority be lowered?

File: Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  [∀ (n : ℤ), (CategoryTheory.shiftFunctor D n).Additive] [Pretriangulated D]
+  [Category A] [Abelian A]
+
+namespace Functor
+
+variable (F : C ⥤ A)
+
+/-- A functor from a pretriangulated category to an abelian category is an homological functor
+if it sends distinguished triangles to exact sequences. -/
+class IsHomological extends F.PreservesZeroMorphisms : Prop where
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean:46]: I think this is usually called "cohomological" in English?
cf: https://ncatlab.org/nlab/show/cohomological+functor

File: Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    obtain ⟨g, rfl⟩ := ex.lift' f (by simpa using hf =≫ biprod.snd)
+    dsimp [S] at hf ⊢
+    replace hf := hf =≫ biprod.fst
+    simp only [assoc, biprodComparison_fst, zero_comp, ← F.map_comp, biprod.inl_fst,
+      F.map_id, comp_id] at hf
+    rw [hf, zero_comp]
+  have : PreservesBinaryBiproduct X₁ X₂ F := preservesBinaryBiproductOfMonoBiprodComparison _
+  apply Limits.preservesBinaryProductOfPreservesBinaryBiproduct
+
+instance [F.IsHomological] : F.Additive := F.additive_of_preserves_binary_products
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean:120]: Should this priority be lowered?

File: Mathlib/CategoryTheory/Triangulated/Functor.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/Triangulated/Functor.lean:125]: Thanks for the suggestion. I am lowering the priority of these instances.

File: Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean (hunk 2 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean:50]: I have added a terminological note in the docstring of the file.

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**